iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerLanguageTest.java

```diff
@@ -19,15 +19,59 @@
  */
 package org.sonar.iac.docker.plugin;
 
+import java.nio.file.Path;
+import java.util.Arrays;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.sonar.api.batch.fs.internal.PathPattern;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 class DockerLanguageTest {
 
   @Test
-  void should_return_docker_file_suffixes() {
+  void shouldReturnDockerFileSuffixes() {
     DockerLanguage language = new DockerLanguage();
     assertThat(language.getFileSuffixes()).isEmpty();
   }
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+    "Dockerfile",
+    "filename.dockerfile",
+    "filename.Dockerfile",
+    "filename.dOckerFilE"
+  })
+  void fileNameIsAssignedToLanguage(String fileName) {
+    assertThat(associatedToLanguage(fileName)).isTrue();
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+    "dockerfile",
+    "somefile",
+    "FooDockerfile",
+    "DockerfileFoo",
+    "Dockerfile.java",
+    "Helloworld.java",
+    // TODO: SONARIAC-1095 Extend DockerLanguage to include possible conflicting filePatterns
+    "Dockerfile.foo"
+  })
+  void fileNameIsNotAssignedToLanguage(String fileName) {
+    assertThat(associatedToLanguage(fileName)).isFalse();
+  }
+
+  private static boolean associatedToLanguage(String fileName) {
+    DockerLanguage language = new DockerLanguage();
+
+    // Based on the actual implementation in SQ and SC
```

Comment:
Is it something from `sonar-plugin-api-impl`? I suggest to provide more details on where the implementation is to make it easier to adapt to potential changes in there

Reply:
No, its from sonarqube and sonarcloud directly, its in the `LanguageDetection` class. I will provide more details.